azure/Guidelines.md

```diff
@@ -846,110 +847,259 @@ https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#perfo
 
-In rare instances where an operation may take a _very long_ time to complete, e.g. longer than 15 minutes,
-it may be better to expose this as a first class resource of the API rather than as an operation on another resource.
+<a href="#lro-no-patch-lro" name="lro-no-patch-lro">:no_entry:</a> **DO NOT** implement PATCH as an LRO. If LRO update semantics are required, implement it using the [LRO POST action pattern](#lro-existing-resource) .
 
-There are two basic patterns for long-running operations in Azure. The first pattern is used for a POST and DELETE
-operations that initiate the LRO. These return a `202 Accepted` response with a JSON status monitor in the response body.
-The second pattern applies only in the case of a PUT operation to create a resource that also involves additional long-running processing.
-For guidance on when to use a specific pattern, please refer to [Considerations for Service Design, Long Running Operations](./ConsiderationsForServiceDesign.md#long-running-operations).
-These are described in the following two sections.
+#### Patterns to Initiate a Long-Running Operation
 
-#### POST or DELETE LRO pattern
+<a href="#lro-valid-inputs-synchronously" name="lro-valid-inputs-synchronously">:white_check_mark:</a> **DO** perform as much validation as practical when initiating an LRO operation to alert clients of errors early.
 
-A POST or DELETE long-running operation accepts a request from the client to initiate the operation processing and returns
-a [status monitor](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.3) that reports the operation's progress.
+<a href="#lro-returns-operation-location" name="lro-returns-operation-location">:white_check_mark:</a> **DO** include an `operation-location` response header with the absolute URL of the status monitor for the operation.
 
-<a href="#lro-no-post-create" name="lro-no-post-create">:no_entry:</a> **DO NOT** use a long-running POST to create a resource -- use PUT as described below.
+<a href="#lro-operation-location-includes-api-version" name="lro-operation-location-includes-api-version">:ballot_box_with_check:</a> **YOU SHOULD** include the `api-version` query parameter in the `operation-location` response header with the same version passed on the initial request but expect a client to change the `api-version` value to whatever a new/different client desires it to be.
 
-<a href="#lro-operation-id-request-header" name="lro-operation-id-request-header">:white_check_mark:</a> **DO** allow the client to pass an `Operation-Id` header with an ID for the operation's status monitor.
+<a href="#lro-put-response-headers" name="lro-put-response-headers">:white_check_mark:</a> **DO** include response headers with any additional values needed for a [GET polling request](#lro-poll) to the status monitor (e.g. location).
```

Comment:
I think this line is moved here without change.

However, it is hard for me to understand the example:
> include response headers with any additional values needed for a GET polling request to the status monitor **(e.g. location)**.

I take the "e.g. location" mean the `location` header. But a `location` header seems not mean much to the GET polling request. (the polling URL should be from `operation-location` header that covered in the sentence above.

Do we have a better example? `retry-after` header is also mentioned in later point. `operation-id` header?
If no good example, maybe skip the example.

Reply:
I agree that the example is poorly explained. The idea is that values passed in headers can be used as parameters -- specifically path parameters -- to the status monitor endpoint. There will always be an `operation-id` -- the final path segment -- but there may be other path parameters and headers can be used to specify what values should be used for these.

I don't know that we should go into all that detail in the guidelines doc, but perhaps we should include this in the ConsiderationsForServiceDesign.